src/OpenFeature/EventExecutor.cs

```diff
@@ -147,7 +147,7 @@ private void StartListeningAndShutdownOld(FeatureProvider newProvider, FeaturePr
             if (oldProvider != null && !this.IsProviderBound(oldProvider))
             {
                 this._activeSubscriptions.Remove(oldProvider);
-                oldProvider.GetEventChannel()?.Writer.Complete();
```

Comment:
I think we should do [this](https://github.com/open-feature/dotnet-sdk/pull/393#issuecomment-2715674476), or keep this check.

it's unlikely, but still technically possible that somebody overrides the `GetEventChannel` and returns null here unless we remove the virtual (which is technically breaking, but not in a way I think we need to worry about).

Reply:
@toddbaert Actually, it is impossible to override and return `null` if you have the `nullable` setting enabled. But I agree with you, I prefer to remove virtual.

Reply:
OK. I think we should not mark this breaking. Even if it technically is, in a very strict sense, I truly doubt anyone has overridden this, and if they did, they did something very wrong and risky. I'm going to remove the `!`.